### PR TITLE
Rework xml parser for comments

### PR DESCRIPTION
**Pull request by Honny1**
This PR reworks class ```_XmlParserComments```.

### CODE REVIEW

File: oval_graph/_xml_parser_comments.py [hunk 1]
```diff
@@ -2,38 +2,44 @@
...
     def __init__(self, oval_definitions):
         self.oval_definitions = oval_definitions
 
-    def _create_dict_form_criteria(self, criteria, description):
+    def _create_dict_form_criteria(self, criteria, description=None):
         comments = dict(
-            operator='AND' if criteria.get('operator') is None else criteria.get('operator'),
-            comment=description if criteria.get('comment') is None else criteria.get('comment'),
+            operator=self._get_operator(criteria),
+            comment=self._get_comment(criteria, description),
```
Comment (jan-cerny) [inline oval_graph/_xml_parser_comments.py:13]: What happens if the description is None?

File: oval_graph/_xml_parser_comments.py [hunk 2]
```diff
@@ -2,38 +2,44 @@
...
+        )
+        if criterion.get('definition_ref'):
+            out['extend_definition'] = criterion.get('definition_ref')
+        else:
+            out['value_id'] = criterion.get('test_ref')
+        return out
+
+    def _get_operator(self, criterion):
+        return 'AND' if criterion.get(
+            'operator') is None else criterion.get('operator')
```
Comment (jan-cerny) [inline oval_graph/_xml_parser_comments.py:37]: I'm not a big fan of this. I would prefer something like this:

```suggestion
        operator = criterion.get('operator')
        return 'AND' if operator is None else operator
```

File: oval_graph/_xml_parser_comments.py (hunk 1 above)
Comment (Honny1) [reply oval_graph/_xml_parser_comments.py:13]: Returns ```None``` or ```criterion.get('comment')```.

**Comment by jan-cerny** [comment]
Thanks for the updates, looks good to me.

**Pull request merged**